### ISSUE TITLE
Location edit - allow to center on GPS provided location

### ISSUE THREAD

**Issue by matkoniecz** [commons-app/apps-android-commons#5008]
### What is the user problem or growth opportunity you want to see solved?

User should not be expected to scroll to their current location through half the world

### How do you know that this problem exists today? Why is this important?

I just tried to add location and have up

![Screenshot_20220709-185242](https://user-images.githubusercontent.com/899988/178115470-4aa4dfc5-fc7f-474a-af09-a6d5a857672d.png)

### Who will benefit from it?

People adding location to uploads

**Comment by nicolas-raoul** [comment]
Yes, the first time you are dropped at a random location in San Francisco, but then the app remembers the last location you chose 🙂

Adding a "center on location" button like in Google Maps is a great idea, there is a good probability that it will be nearer to where the picture was taken than San Francisco. Please note that using GPS often results in permission issues, so it might not be trivial to implement&test though.

**Comment by matkoniecz** [comment]
> there is a good probability that it will be nearer to where the picture was taken than San Francisco

and also likely closer than the last remembered location!

**Comment by matkoniecz** [comment]
Feel free to work on it (as far as I am concerned - and it seems that noone else is working on it)

**Comment by russianbear222** [comment]
Hi, will this issue be closed? @mattflaschen

**Comment by nicolas-raoul** [comment]
@pwq334107948 Have you noticed that this issue is already assigned to @russianbear222 ?
Hopefully you two can collaborate (for instance to add unit tests) but in the future please make sure the issue is assigned to you before starting coding 🙂

@russianbear222 Feel free to pull pwq's code and improve it, thanks!

**Comment by russianbear222** [comment]
@nicolas-raoul I added a new pull request, however, I wasn't able to do unit tests, I don't understand the format of it, I get java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at fr.free.nrw.commons.locationpicker.LocationPickerActivityUnitTests.testCenterButton(LocationPickerActivityUnitTests.kt:256)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:568)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:278)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException
	at fr.free.nrw.commons.LocationPicker.LocationPickerActivity.getCentre(LocationPickerActivity.java:444)
	... 22 more,

 although I made a simple test

 @Test
    @Throws(Exception::class)
    fun testCenterButton() {
        val method: Method = LocationPickerActivity::class.java.getDeclaredMethod(
            "getCentre"
        )
        method.isAccessible = true
        method.invoke(activity)
    }

**Comment by russianbear222** [comment]
@nicolas-raoul I just pushed a new pull request.

**Comment by nicolas-raoul** [comment]
Fixed, thanks all!
